fc/parsing/CompactSyntaxParser.py

```diff
@@ -984,6 +984,9 @@ def MonkeyPatch():
 
     def ignore(self, other):
         """Improved ignore that avoids ignoring self by accident."""
+        if isinstance(other, str):
+            other = Suppress(other)
+
         if isinstance(other, p.Suppress):
             if other not in self.ignoreExprs and other != self:
```

Comment:
Maybe making the second test here `other is not self` would suffice for all versions? Actually, no, I suspect we do need the equality check because you could potentially have multiple `Suppress` objects wrapping the same underlying expression.